solutions/gold/usaco-696.mdx

```diff
@@ -99,8 +99,81 @@ public class PromotionCounting {
 }
 ```
 </JavaSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+const int mx = 1e5 + 1;
+int n, t = 0, out[mx], in[mx], bit[mx], ans[mx];
+vector<int> adj[mx];
+
+int sum(int i) {
+	int res = 0;
+	i++;
+	while (i > 0) {
+		res += bit[i];
+		i -= i & -i;
+	}
+	return res;
+}
+
+void update(int i, int v) {
+	i++;
+	while (i <= n) {
+		bit[i] += v;
+		i += i & -i;
+	}
+}
```

Comment:
use a struct for bits
[this](https://usaco.guide/gold/PURS?lang=cpp#solution---dynamic-range-sum-queries-with-a-bit) part has a good impl